src/charm.py

```diff
@@ -105,6 +105,12 @@ def _on_config_changed(self, event: ops.ConfigChangedEvent) -> None:
             self.unit.status = ops.BlockedStatus(str(e))
             event.defer()
             return
+        restart_services = False
+        restart_services = utils.config_change_requires_restart(self.config)
+        if restart_services:
+            logger.info("Begin configuration change: %s", restart_services)
+        else:
+            logger.info("No restart required for config change.")
```

Comment:
I don't get this one. It will always trigger a restart, except when using docker-tag? It will always trigger a restart even though the config was not changed for these parameters, since it only checks if the parameter is set or not, but does not consider if it was changed...